### ISSUE TITLE
Code for analysis of car sales

### ISSUE THREAD

**Issue by neokaamble11** [neokaamble11/carsalesanalysis#1]
import pandas as pd

df=pd.read_csv("car_price_dataset.csv")

print(df.isnull().sum())

df=df.drop_duplicates()

df=df[df['Mileage']<=500000]

avg_price_by_brand = df.groupby('Brand')['Price'].mean().sort_values(ascending=False)

popular_fuel_type = df['Fuel_Type'].value_counts().idxmax()

avg_mileage_by_year = df.groupby('Year')['Mileage'].mean()

correlation = df['Mileage'].corr(df['Price'])

import matplotlib.pyplot as plt

import seaborn as sns

sns.barplot(x=avg_price_by_brand.index, y=avg_price_by_brand.values)
plt.xticks(rotation=90)
plt.title('Average Price by Brand')
plt.show()

sns.scatterplot(x='Mileage', y='Price', data=df)
plt.title('Mileage vs Price')
plt.show()

fuel_counts = df['Fuel_Type'].value_counts()

print(df.columns)

plt.figure(figsize=(8,8))
plt.pie(fuel_counts,labels=fuel_counts.index,autopct='%1.1f%%',startangle=140,colors=['lightblue','lightgreen','lightcoral','gold','purple'])
plt.title('Distribution of Cars by Fuel Type')
plt.show()

df['Engine_Size'].hist(bins=10, color='green', edgecolor='black')
plt.xlabel('Engine Size')
plt.ylabel('Frequency')
plt.title('Distribution of Engine Sizes')
plt.show()

df['Brand'].value_counts().head(10).plot(kind='bar', color='cyan')
plt.xlabel('Brand')
plt.ylabel('Number of Cars')
plt.title('Top 10 Most Common Car Brands')
plt.show()

df['Transmission'].value_counts().plot(kind='bar', color=['skyblue', 'salmon'])
plt.xlabel('Transmission Type')
plt.ylabel('Number of Cars')
plt.title('Transmission Type Popularity')
plt.show()

avg_price_year = df.groupby('Year')['Price'].mean()
avg_price_year.plot(kind='line', marker='o', color='purple')
plt.xlabel('Year')
plt.ylabel('Average Price')
plt.title('Average Car Price Over the Years')
plt.grid()
plt.show()

df['Price_per_1000_miles'] = df['Price'] / (df['Mileage'] / 1000)
df.sort_values('Price_per_1000_miles', ascending=True).head(10)

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LinearRegression
from sklearn.metrics import mean_absolute_error, mean_squared_error, r2_score
df=pd.read_csv('car_price_dataset.csv')
print(df.info())

df=df.dropna()

features = ['Mileage','Engine_Size','Year']
target='Price'
for col in features +[target]:
    if col not in df.columns:
        raise KeyError(f"Column '{col}' not found in dataset")
X=df[features]
y=df[target]
X_train, X_test, y_train, y_test = train_test_split(X,y,test_size=0.2,random_state=42)
model=LinearRegression()
model.fit(X_train, y_train)
y_pred=model.predict(X_test)
mae = mean_absolute_error(y_test, y_pred)
mse = mean_squared_error(y_test, y_pred)
rmse = np.sqrt(mse)
r2 = r2_score(y_test, y_pred)
print(f"Mean Absolute Error (MAE): {mae:.2f}")
print(f"Root Mean Squared Error (RMSE): {rmse:.2f}")
print(f"R-squared (R²) Score: {r2:.2f}")
plt.figure(figsize=(8,6))
sns.scatterplot(x=y_test, y=y_pred, alpha=0.7, color='blue')
plt.plot([min(y_test), max(y_test)], [min(y_test), max(y_test)], linestyle='--', color='red')  # 45-degree line
plt.xlabel("Actual Prices")
plt.ylabel("Predicted Prices")
plt.title("Actual vs Predicted Car Prices")
plt.show()
